src/main/java/org/mule/extension/sftp/internal/connection/SftpClient.java

```diff
@@ -146,6 +152,15 @@ public SftpClient(String host, int port, PRNGAlgorithm prngAlgorithm, SchedulerS
     }
   }
 
+  /**
+   * Contains the code to configure / overwrite crypto factories required during creation of {@link SshClient}
+   * If the externalConfigs provided doesn't contain a particular factory or crypto algo, then it will use the default.
+   * @param externalConfigProvider Provides the way to inject external configuration
+   */
+  private void configureWithExternalSources(ExternalConfigProvider externalConfigProvider) {
+    Properties properties = externalConfigProvider.getConfigProperties();
```

Comment:
Isn't there any code path where `externalConfigProvider` is `null`?

Reply:
We haven't created that path. Wherever there is no requirement of passing this provider we are only providing 
() -> new Properties()